okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -493,14 +493,21 @@ public void close() {}
           if (proxy == null) {
             sock = new Socket(address.getAddress(), address.getPort());
           } else {
-            sock = createHttpProxySocket(
-                address,
-                proxy.getProxyAddress(),
-                proxy.getUsername(),
-                proxy.getPassword()
-            );
+            if (proxy.getProxyAddress() == null) {
+              // This should never happen, case added for readability
+              throw new AssertionError();
+            } else if (proxy.getProxyAddress() instanceof InetSocketAddress) {
+              sock = createHttpProxySocket(
+                  address,
+                  (InetSocketAddress)proxy.getProxyAddress(),
+                  proxy.getUsername(),
+                  proxy.getPassword()
+              );
+            } else {
+              throw new Error(
```

Comment:
This should not be an error. Given the catch that's a bit lower, you can do `throw Status.INTERNAL.withDescription("Unsupported SocketAddress implementation " + proxy.getProxyAddress().getClass()).asException();`